### PR TITLE
Add textfield functionality

### PR DESCRIPTION
**Pull request by ghost**
Added text field functionality, as suggested in #66

### CODE REVIEW

**Comment by syky27** [comment]
Hi @Luukje thank you for the contribution, but the diff is too large could you ensure that you change only things that need to be changed?

Thank you

**Comment by syky27** [comment]
@Luukje  bump?

**Comment by Luukje** [comment]
Yes, I only changed things that need to be changed. 🙂

File: JSSAlertView/Classes/JSSAlertView.swift [hunk 1]
```diff
@@ -18,138 +18,144 @@ import UIKit
...
 public enum TextColorTheme {
-	case dark, light
+    case dark, light
 }
 
 
 /// Custom modal controller
 open class JSSAlertView: UIViewController {
-
-	var containerView: UIView!
```
Comment (syky27) [inline JSSAlertView/Classes/JSSAlertView.swift]: here

File: JSSAlertView/Classes/JSSAlertView.swift [hunk 2]
```diff
@@ -18,138 +18,144 @@ import UIKit
 /// - dark: Dark Text Color theme
 /// - light: Light Text Color theme
 public enum TextColorTheme {
-	case dark, light
```
Comment (syky27) [inline JSSAlertView/Classes/JSSAlertView.swift]: here

File: JSSAlertView/Classes/JSSAlertView.swift [hunk 3]
```diff
@@ -18,138 +18,144 @@ import UIKit
...
+    var closeActionWithTextField: ((String?)->Void)!
+    var closeAction: (()->Void)!
+    var cancelAction: (()->Void)!
+    var isAlertOpen: Bool = false
+    var noButtons: Bool = false
     
     var timeLeft: UInt?
-
-	enum FontType {
-		case title, text, button, timer
```
Comment (syky27) [inline JSSAlertView/Classes/JSSAlertView.swift]: here

File: JSSAlertView/Classes/JSSAlertView.swift [hunk 4]
```diff
@@ -18,138 +18,144 @@ import UIKit
...
-
-	var darkTextColor = UIColorFromHex(0x000000, alpha: 0.75)
-	var lightTextColor = UIColorFromHex(0xffffff, alpha: 0.9)
-
-	public enum ActionType {
-		case close, cancel
-	}
-
-	let baseHeight: CGFloat = 160.0
-	var alertWidth: CGFloat = 290.0
```
Comment (syky27) [inline JSSAlertView/Classes/JSSAlertView.swift]: here

File: JSSAlertView/Classes/JSSAlertView.swift [hunk 5]
```diff
@@ -18,138 +18,144 @@ import UIKit
...
-		self.viewHeight = size.height
-
-		var yPos: CGFloat = 0.0
-		let contentWidth:CGFloat = self.alertWidth - (self.padding*2)
-
-		// position the icon image view, if there is one
-		if self.iconImageView != nil {
-			yPos += iconImageView.frame.height
-			let centerX = (self.alertWidth-self.iconImageView.frame.width)/2
-			self.iconImageView.frame.origin = CGPoint(x: centerX, y: self.padding)
```
Comment (syky27) [inline JSSAlertView/Classes/JSSAlertView.swift]: here

**Review by syky27 — CHANGES_REQUESTED** [review]
Please look at this diff, and fix it. https://github.com/JSSAlertView/JSSAlertView/pull/79/commits/a57f11d0a5708115944071f5ad04e6b0b981463f

**Comment by syky27** [comment]
@Luukje You most definetely did not changed only things that needed to be changed....

**Comment by syky27** [comment]
Are you willing to tidy up the git?

**Comment by mohsinka** [comment]
Any update on this?

**Comment by syky27** [comment]
@mohsinka Not from my side, If you are willing you can tidy up the diff... :)

Sorry but I am currently unavailable to do so...

**Comment by syky27** [comment]
@mohsinka Hey, I got around to do this. I had to do in other branch because the previous has been deleted. There will be new version with this functionality `5.0.0` shortly.

**Pull request closed without merging**